### PR TITLE
[Serve] Ensure serve shutdown are not spamming logs

### LINKED ISSUE #41013: [Serve] serve shutdown not to log multiple shutdown started and application deletion messages
We don't want to keep spamming messages like below. The messages should only be logged once

```
INFO 2023-11-07 14:24:47,773 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:47,776 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:47,887 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:47,887 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:47,996 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,001 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,113 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,117 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,232 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,237 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,347 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,347 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,452 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,453 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,582 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,585 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,691 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,698 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,805 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,814 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,923 controller 96971 controller.py:517 - Controller shutdown started!
INFO 2023-11-07 14:24:48,935 controller 96971 application_state.py:265 - Deleting application 'default'
INFO 2023-11-07 14:24:48,941 controller 96971 deployment_state.py:2034 - Replica default#A#uGWtrA is stopped.
```

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

Currently during serve shutdown process, each shutdown call will produce a log message to the log files util all resources are completely shut down. This PR fixes it by logging it only once at the first time when the shutdown is called. 

## Related issue number

Closes: https://github.com/ray-project/ray/issues/41013

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_standalone_3.py [hunk 1]
```diff
@@ -610,5 +611,40 @@ def __call__(self):
     serve.shutdown()
 
 
+def test_serve_shutdown_without_duplicated_logs(shutdown_ray, call_ray_stop_only):
+    """Test Serve shutdown without duplicated logs.
```
Comment (zcin) [inline python/ray/serve/tests/test_standalone_3.py:615]:
```suggestion
    """Test Serve shuts down without duplicated logs.
```

**Review by zcin — APPROVED** [review]
Thanks!

File: python/ray/serve/_private/controller.py [hunk 2]
```diff
@@ -511,8 +511,8 @@ def shutdown(self):
 
         if self._shutdown_start_time is None:
             self._shutdown_start_time = time.time()
+            logger.info("Controller shutdown started!", extra={"log_to_stderr": False})
```
Comment (edoakes) [inline python/ray/serve/_private/controller.py:514]: Remove the exclamation point please (we should basically never have these in logs)

**Review by edoakes — APPROVED** [review]

**Comment by GeneDer** [comment]
@architkulkarni Can you help to merge this? The failing tests are flaky and unrelated 
image

**Pull request merged**